src/main/java/seedu/todo/model/TodoList.java

```diff
@@ -171,13 +127,26 @@ public void load(String location) throws ValidationException {
     }
 
     @Override
-    public String getStorageLocation() {
-        return storage.getLocation();
+    public void setTasks(List<ImmutableTask> todoList) {
+        setTasks(todoList, true);
+    }
+
+    /**
+     * We have a private version of setTasks because we also need to setTask during initialization, 
+     * but we don't want the list to be save during init (where we presumably got the data from)
+     */
+    private void setTasks(List<ImmutableTask> todoList, boolean persistToDisk) {
```

Comment:
`isPersistedToDisk` is wrong, because `is` prefix is for variables with state (it would imply that the data *has already been* persisted to disk). I guess you *could* replace this with an `enum`, but I think the semantics of the boolean as a flag is okay